### ISSUE TITLE
fix(model): make Session.Generate handle empty prompts without panic

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12605]
# fix(model): make Session.Generate handle empty prompts without panic

```routing
lane: model
paths: ["internal/model/kv.go", "internal/model/audit_generate_empty_prompt_test.go"]
expected_steps: 4
```

## Current state

At source SHA `6872f998b465d96633c3356ef2382c97fc98cc7c`, `Session.Generate(nil, 1)` panics. `Generate` calls `Prefill(prompt)`, `Prefill` intentionally returns `nil` for an empty prompt, and `Generate` immediately passes that nil logits slice to `argmaxF32`, which indexes `v[0]`.

Duplicate search status: local ticket/docs search and parent GitHub search found no exact existing issue for `Session.Generate` empty-prompt panic.

Relevant seams:

- `internal/model/kv.go:1008-1011` (`Session.Prefill`) returns nil for empty prompts.
- `internal/model/kv.go:1287-1297` (`Session.Generate`) does not guard empty prompts before decoding.
- `internal/model/kv.go:1334-1341` (`argmaxF32`) assumes a non-empty logits slice and indexes `v[0]`.

## Working spine

Add the smallest explicit empty-prompt branch at `Session.Generate` before it calls `argmaxF32`, then pin nil and zero-length prompts with a regression test.

## Parent context

Independent inference-correctness audit leaf from the 2026-09-09 model generation helper review.

## Why this is next

The exported greedy generation helper currently turns `Prefill`'s empty-input nil return into an untyped panic instead of explicit empty-input behavior.

## Core through-line

The exported greedy generation helper should define a safe empty-prompt behavior. It should not convert `Prefill`'s nil sentinel into an untyped slice-bounds panic.

## In scope

- `Session.Generate` behavior for nil and zero-length prompts.
- A direct regression test for the exported helper.
- Preservation of current non-empty prompt generation behavior.

## Gold-plating boundary

Do not redesign generation APIs, add BOS-token insertion, or change backend/session cache behavior in this ticket. Keep scope to `Session.Generate` and a direct regression test. Batch generation and context-returning APIs can be audited separately if needed.

## Out of scope

Batch generation, BOS synthesis, and context-returning generation API policy are out of scope.

## Implementation steps

1. Add an entry guard in `Session.Generate` for `len(prompt) == 0` before calling `Prefill`.
2. Choose the smallest behavior that preserves the current signature, preferably returning no generated tokens for empty prompts.
3. Add `internal/model/audit_generate_empty_prompt_test.go` covering both nil and zero-length prompts.
4. Run the empty-prompt witness and existing non-empty prompt generation tests.

## Done condition

- [ ] `Session.Generate(nil, n)` does not panic.
- [ ] `Session.Generate([]int{}, n)` does not panic.
- [ ] The chosen behavior is explicit in code and tests; returning no generated tokens is acceptable without changing the exported signature.
- [ ] Existing non-empty prompt generation tests still pass.

## Definition of Done

Done when empty prompts have explicit `Session.Generate` behavior, both nil and zero-length prompt witnesses pass, and non-empty generation coverage remains green.

## Acceptance gate

`go test ./internal/model -run TestAuditGenerateRejectsEmptyPromptInsteadOfPanicking -count=1` passes after the fix, and existing non-empty prompt generation tests still pass.

## Closure binding

Resolving commit cites the created GitHub issue and carries `(fak model)`.

## Witness

Captured failure from the audit overlay:

```text
$ go test ./internal/model -run TestAuditGenerateRejectsEmptyPromptInsteadOfPanicking -count=1
--- FAIL: TestAuditGenerateRejectsEmptyPromptInsteadOfPanicking (0.00s)
    audit_repro_test.go:144: Generate(nil, 1) panicked instead of rejecting empty prompt: runtime error: index out of range [0] with length 0
FAIL
FAIL    github.com/anthony-chaudhary/fak/internal/model    0.024s
```

## Verifiable Witness

Place this standalone test in `internal/model/audit_generate_empty_prompt_test.go`, then run:

```bash
go test ./internal/model -run TestAuditGenerateRejectsEmptyPromptInsteadOfPanicking -count=1
```

```go
package model

import "testing"

func auditGenerateEmptyTinyConfig() Config {
	return Config{
		ModelType:         "llama",
		VocabSize:         16,
		HiddenSize:        32,
		IntermediateSize:  64,
		NumLayers:         1,
		NumHeads:          2,
		NumKVHeads:        2,
		HeadDim:           16,
		RMSNormEps:        1e-6,
		RopeTheta:         10000,
		TieWordEmbeddings: true,
		EOSTokenID:        -1,
	}
}

func captureGenerateEmptyPanic(fn func()) (any, bool) {
	var value any
	ok := false
	func() {
		defer func() {
			if value = recover(); value != nil {
				ok = true
			}
		}()
		fn()
	}()
	return value, ok
}

func TestAuditGenerateRejectsEmptyPromptInsteadOfPanicking(t *testing.T) {
	m := NewSynthetic(auditGenerateEmptyTinyConfig())
	for _, prompt := range [][]int{nil, []int{}} {
		s := m.NewSession()
		panicValue, panicked := captureGenerateEmptyPanic(func() {
			_ = s.Generate(prompt, 1)
		})
		s.Close()
		if panicked {
			t.Fatalf("Generate(%v, 1) panicked instead of handling an empty prompt: %v", prompt, panicValue)
		}
	}
}
```

## Likely files

- `internal/model/kv.go`
- `internal/model/audit_generate_empty_prompt_test.go`

## Lane

model

## Work unit

leaf

## Expected steps

4

## Trigger

Inference audit witness captured on 2026-09-09.

## Batch policy

One GitHub issue per `fak-model-key`; update this issue if another empty-prompt panic reproduction lands for the same `Session.Generate` seam.

## Problem frame

- Centrality: Core
- P1 Context: preserved - concrete exported helper panic at source SHA `6872f998b465d96633c3356ef2382c97fc98cc7c`; no duplicate found in local model tickets/tests.
- P2 Net value: advanced - replaces an avoidable runtime panic with deterministic empty-input behavior.
- P3 Adaptation: preserved - non-empty prompt generation semantics remain unchanged.
- P4 Operations: preserved - one package, two likely files, four implementation steps.

## Blast radius / fallback

Blast radius is limited to `Session.Generate` on empty input. If returning an empty result conflicts with an existing caller, the fallback is to keep `Prefill` unchanged and add the smallest explicit guard at `Generate`'s entry point.
